bon/tests/integration/builder/attr_skip.rs

```diff
@@ -1,72 +1,27 @@
 use crate::prelude::*;
 use expect_test::expect;
 
-#[cfg(feature = "alloc")]
-#[test]
-fn fn_alloc() {
-    #[builder]
-    fn sut(
-        #[builder(skip = "skip".to_owned())] arg1: String,
-        #[builder(skip = vec![42])] arg2: Vec<u32>,
-    ) -> (String, Vec<u32>) {
-        (arg1, arg2)
-    }
-
-    assert_debug_eq(sut().call(), expect![[r#"("skip", [42])"#]]);
-}
-
 #[cfg(feature = "alloc")]
 #[test]
 fn struct_alloc() {
-    use bon::bon;
     use expect_test::expect;
 
     #[builder]
     #[derive(Debug)]
     struct Sut {
         #[builder(skip = "skip".to_owned())]
+        #[allow(dead_code)]
         arg1: String,
 
         #[builder(skip = vec![42])]
+        #[allow(dead_code)]
         arg2: Vec<u32>,
     }
 
     assert_debug_eq(
         Sut::builder().build(),
         expect![[r#"Sut { arg1: "skip", arg2: [42] }"#]],
     );
```

Comment:
It might be good to add a [compile fail](https://doc.rust-lang.org/stable/rustdoc/write-documentation/documentation-tests.html#attributes) test which asserts that trying to call the setter of a skipped field or trying to explicitly set a skipped field runs into error.

Reply:
Hi! Thank you for the suggestions! I applied them in the commit 727ad96fca557279db0da5829050a195a3661ecb in the other PR https://github.com/elastio/bon/pull/70.

However, I'd like to note several things:

I try to write the tests in this crate to be `no_std` compatible, so I adapted your suggestions to avoid using `Vec` and `String` which would require `alloc` being available, and replaced them with `u32` and `[u32; 3]`. This will make sure the code works when `std` feature of the crate is disabled. I don't think a test using `String/Vec` would catch any more bugs in this case, simple `u32`s will suffice.

---

I added the test for the lack of setters for skipped members to `ui/compile_fail/misc.rs` file, which is covered by [`trybuild`](https://docs.rs/trybuild/latest/trybuild/) and it also makes it possible to see the exact compile error message produced in the snapshot (`misc.stderr`)

---

To be clear, there are existing tests for this feature, but they were rather simple. These are the doc tests from the examples on the website ([here for `skip`](https://github.com/elastio/bon/blob/029cf450a6d10dfdf0a8916d7944623d79ed117d/website/reference/builder.md?plain=1#L844-L892), and [here for `default`](https://github.com/elastio/bon/blob/029cf450a6d10dfdf0a8916d7944623d79ed117d/website/reference/builder.md?plain=1#L572-L651)). I have a script in `e2e-tests` directory that generates Rust files with doc tests for every markdown file in the `website` directory, so all code snippets on the website are actually tested:
  https://github.com/elastio/bon/blob/029cf450a6d10dfdf0a8916d7944623d79ed117d/e2e-tests/build.rs#L1-L3